### ISSUE TITLE
load library error on macos 10.12 with R 3.4.3

### ISSUE THREAD

**Issue by deschsimon** [eddelbuettel/littler#56]
I' using macOS 10.12 and have just recently upgraded my R version from 3.3.3 to 3.4.3. 
```r
> sessionInfo()
R version 3.4.3 (2017-11-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] de_DE.UTF-8/de_DE.UTF-8/de_DE.UTF-8/C/de_DE.UTF-8/de_DE.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] littler_0.3.3

loaded via a namespace (and not attached):
[1] compiler_3.4.3 tools_3.4.3    yaml_2.1.16
```
I have then re-linked the r binary `/Library/Frameworks/R.framework/Versions/3.4/Resources/library/littler/bin/r` to `/usr/local/bin/lr`.
Since I get the error 
```bash
dyld: Library not loaded: /usr/local/clang4/lib/libomp.dylib
 Referenced from: /usr/local/bin/lr
  Reason: image not found
Abort trap: 6
```
when running R code using r.
The library is located at `/Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libomp.dylib`. So I assume that some PATH is not beeing linked correctly.

I have tried to remove and reinstall the package as well as updating with no luck.
I have then tried to compile the package from source as described for versions prior to 0.2.3 but this didn't work.
If I link the Current R version to 3.3 and link `lr`to the 3.3 package binary everything works fine again.

Any help on getting this issue solved is highly appreciated!
Thanks in advance!

**Comment by eddelbuettel** [comment]
What happens if you simply try to reinstall [littler](https://cran.r-project.org/package=littler) from CRAN?   Sometimes the toolchain changes along with R and you may have to recompile.

**Comment by deschsimon** [comment]
Thanks for the quick response!

I tried
```r
> remove.packages('littler')
Removing package from ‘/Library/Frameworks/R.framework/Versions/3.4/Resources/library’
(as ‘lib’ is unspecified)
> install.packages('littler')
versuche URL 'https://cran.rstudio.com/bin/macosx/el-capitan/contrib/3.4/littler_0.3.3.tgz'
Content type 'application/x-gzip' length 385494 bytes (376 KB)
==================================================
downloaded 376 KB


The downloaded binary packages are in
	/var/folders/yv/f07tn6d90g161by4mfg1hblm0000gp/T//RtmpqwHfux/downloaded_packages
```
with no luck.
I have also tried to install from github:
```r
> remove.packages('littler')
> install_github('eddelbuettel/littler')
Downloading GitHub repo eddelbuettel/littler@master
from URL https://api.github.com/repos/eddelbuettel/littler/zipball/master
Installing littler
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/yv/f07tn6d90g161by4mfg1hblm0000gp/T/RtmpqwHfux/devtools61461c70503/eddelbuettel-littler-5537245'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library' --install-tests 

* installing *source* package ‘littler’ ...
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang accepts -g... yes
checking for clang option to accept ISO C89... none needed
checking how to run the C preprocessor... clang -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking getopt.h usability... yes
checking getopt.h presence... yes
checking for getopt.h... yes
checking whether sys/types.h defines makedev... yes
checking for setenv... yes
checking for gettimeofday... yes
checking for time... yes
checking if R was built as a shared library... yes
checking for install_name_tool... /usr/bin/install_name_tool
configure: creating ./config.status
config.status: creating src/Makevars
config.status: creating src/gitversion.h
config.status: creating src/config.h
** libs
clang -o r -Wall -g -O2 -I/Library/Frameworks/R.framework/Resources/include littler.c  -L/usr/local/lib -fopenmp -L/usr/local/lib -F/Library/Frameworks/R.framework/.. -framework R -lpcre -llzma -lbz2 -lz -licucore -lm -liconv -L/Library/Frameworks/R.framework/Resources/lib -lRblas -L/Library/Frameworks/R.framework/Resources/lib -lRlapack 
clang: error: unsupported option '-fopenmp'
clang: error: unsupported option '-fopenmp'
make: *** [r] Error 1
ERROR: compilation failed for package ‘littler’
* removing ‘/Library/Frameworks/R.framework/Versions/3.4/Resources/library/littler’
Installation failed: Command failed (1)
```
Due to the hint in the README II.3 I tried to compile from source which didn't work out either.

Any other idea?

**Comment by eddelbuettel** [comment]
I am sorry that you have to go through this but on macOS there is _general mess-up_ with R, the compiler, and OpenMP that littler has nothing to do with.  In essence, R things it has OpenMP when it doesn't. So it creates a faulty `src/Makevars`.  You _could_ edit your `$RHOME/etc/Makeconf` and remove the `-fopenmp` reference R is not truthful about.  But I am not a macOS user so ...  maybe @coatless can help.

**Comment by coatless** [comment]
Grab the `clang4` binary installer from: https://github.com/coatless/r-macos-clang
Install it. This loads and setups the paths for the `clang4` binary used by CRAN.

**Comment by deschsimon** [comment]
@eddelbuettel Thank you for forwarding the issue! I knew it was a macOS related problem since r running fine with R 3.4.3 on my Ubuntu server...

@coatless Great! Thank this solved the issue!

**Comment by Guilz** [comment]
Thank you for this fix @coatless!
Any chance to have this fix incorporated in `littler`, so that it works right away after `install.packages()`?

**Comment by eddelbuettel** [comment]
Come again?  How can the _littler source package_ incorporate what you need to install on your box?

**Comment by Guilz** [comment]
Sorry, I realized this is impossible soon after I wrote the comment... Let me rephrase: do you think there is any chance that this clang4 could be included into the macOS standard R installation, if enough users requested this appropriately to the R core developers?

**Comment by eddelbuettel** [comment]
That I do not know, as I don't own macOS machines.  But dear @coatless has tried and tried and tried again to get that change to happen ... and is AFAIK still waiting.  Not sure what you can do besides lobbying for it on the r-sig-mac list.  Sorry!

**Comment by eddelbuettel** [comment]
This seems ready to be closed.
